src/main/java/youyihj/zenutils/api/config/elements/ConfigGroup.java

```diff
@@ -0,0 +1,419 @@
+    }
+
+    @ZenMethod
+    public ConfigMap intArrayMap(String name, @Optional Map<String, Integer[]> defaultVal) {
+        return addChild(new ConfigMap(this, name, Integer[].class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap stringArrayMap(String name, @Optional Map<String, String[]> defaultVal) {
+        return addChild(new ConfigMap(this, name, String[].class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap booleanMap(String name, @Optional Map<String, Boolean> defaultVal) {
+        return addChild(new ConfigMap(this, name, Boolean.class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap doubleMap(String name, @Optional Map<String, Double> defaultVal) {
+        return addChild(new ConfigMap(this, name, Double.class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap floatMap(String name, @Optional Map<String, Float> defaultVal) {
+        return addChild(new ConfigMap(this, name, Float.class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap intMap(String name, @Optional Map<String, Integer> defaultVal) {
+        return addChild(new ConfigMap(this, name, Integer.class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigMap stringMap(String name, @Optional Map<String, String> defaultVal) {
+        return addChild(new ConfigMap(this, name, String.class, defaultVal));
+    }
+
+    @ZenMethod
+    public ConfigEnum configEnum(String name, String defaultValue, String[] enums) {
+        return addChild(new ConfigEnum(this, name, defaultValue, enums));
+    }
+
+
+    public String getClassName() {
+        String className = "dynamic.zenutils.config." + this.getAbsoluteName();
+        {
+            int idx = className.lastIndexOf('.') + 1;
+            if (idx > 0 && idx < className.length()) {
+                className = className.substring(0, idx) +
+                        Character.toUpperCase(className.charAt(idx)) +
+                        className.substring(idx + 1);
+            }
+        }
+        return className;
+    }
+
+    public Set<String> getClasses() {
+        HashSet<String> hashSet = new HashSet<>();
+        for (ConfigElement element : this.children) {
+            if (element instanceof ConfigGroup) {
+                hashSet.addAll(((ConfigGroup)element).getClasses());
+            }
+        }
+        hashSet.add(this.getClassName());
+        return hashSet;
+    }
+
+    @ZenMethod
+    public String register() {
+        register0();
+        if (this.parent == null) {
+            try {
+                for (String clsN : this.getClasses()) {
+                    CraftTweakerAPI.registerClass(Class.forName(clsN));
+                }
+                return ConfigAnytimeAnytime.register(Class.forName(getClassName()));
+            } catch (Throwable t) {
+                throw new RuntimeException(t);
+            }
+        } else return "";
+    }
+
+    private static class ConfigAnytimeAnytime {
```

Comment:
Anytime² :concerned:
Looks like you copied configanytime, remember to retain the origin copyright notice.